src/Stratis.Bitcoin.Features.Consensus/ConsensusLoop.cs

```diff
@@ -481,22 +480,18 @@ internal void ValidateAndExecuteBlock(ContextInformation context)
         /// This method try to load from cache the UTXO of the next block in a background task. 
         /// </summary>
         /// <param name="flags">Information about activated features.</param>
-        /// <returns>The process task.</returns>
-        private Task TryPrefetchAsync(DeploymentFlags flags)
+        private async void TryPrefetchAsync(DeploymentFlags flags)
         {
             this.logger.LogTrace("({0}:{1})", nameof(flags), flags);
 
-            Task prefetching = Task.FromResult<bool>(true);
-
             if (this.UTXOSet is CachedCoinView)
             {
                 Block nextBlock = this.Puller.TryGetLookahead(0);
                 if (nextBlock != null)
-                    prefetching = this.UTXOSet.FetchCoinsAsync(GetIdsToFetch(nextBlock, flags.EnforceBIP30));
+                    await this.UTXOSet.FetchCoinsAsync(GetIdsToFetch(nextBlock, flags.EnforceBIP30)).ConfigureAwait(false);
```

Comment:
We don't want to await here, this should fire and forget.

Reply:
the logic you are looking for is there, just not here

the only one who calls `TryPrefetchAsync` is `ValidateAndExecuteBlock`:

```
            // Attempt to load into the cache the next set of UTXO to be validated.
            // The task is not awaited so will not stall main validation process.
            this.TryPrefetchAsync(context.Flags);
```

in this code we do not await and hence `TryPrefetchAsync` is `async void`
but inside `TryPrefetchAsync` we want await as that is no longer the point of entry to the new async task

so it works as you want it to and this is correct i believe

Reply:
ah right that's fine.